tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,500 @@
+# States Department of Commerce. This software was developed in part by employees of
+# NIST and in part by NIST contractors. Copyright in portions of this software that
+# were developed by NIST contractors has been licensed or assigned to NIST. Pursuant
+# to Title 17 United States Code Section 105, works of NIST employees are not
+# subject to copyright protection in the United States. However, NIST may hold
+# international copyright in software created by its employees and domestic
+# copyright (or licensing rights) in portions of software that were assigned or
+# licensed to NIST. To the extent that NIST holds copyright in this software, it is
+# being made available under the Creative Commons Attribution 4.0 International
+# license (CC BY 4.0). The disclaimers of the CC BY 4.0 license apply to all parts
+# of the software developed or licensed by NIST.
+#
+# ACCESS THE FULL CC BY 4.0 LICENSE HERE:
+# https://creativecommons.org/licenses/by/4.0/legalcode
+"""Test suite for queue operations.
+
+This module contains a set of tests that validate the CRUD operations and additional
+functionalities for the queue entity. The tests ensure that the queues can be
+registered, renamed, deleted, and locked/unlocked as expected through the REST API.
+"""
+from __future__ import annotations
+
+from typing import Any
+
+import pytest
+from flask.testing import FlaskClient
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.test import TestResponse
+
+from dioptra.restapi.routes import V1_QUEUES_ROUTE, V1_ROOT
+
+from ..lib import actions, helpers
+
+# -- Actions ---------------------------------------------------------------------------
+
+
+def modify_queue(
+    client: FlaskClient,
+    queue_id: int,
+    new_name: str,
+    new_description: str,
+) -> TestResponse:
+    """Rename a queue using the API.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to rename.
+        new_name: The new name to assign to the queue.
+        new_description: The new description to assign to the queue.
+
+    Returns:
+        The response from the API.
+    """
+    payload = {"name": new_name, "description": new_description}
+
+    return client.put(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}/{queue_id}",
+        json=payload,
+        follow_redirects=True,
+    )
+
+
+def delete_queue_with_id(
+    client: FlaskClient,
+    queue_id: int,
+) -> TestResponse:
+    """Delete a queue using the API.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to delete.
+
+    Returns:
+        The response from the API.
+    """
+
+    return client.delete(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}/{queue_id}",
+        follow_redirects=True,
+    )
+
+
+# -- Assertions ------------------------------------------------------------------------
+
+
+def assert_queue_response_contents_matches_expectations(
+    response: dict[str, Any], expected_contents: dict[str, Any]
+) -> None:
```

Comment:
Missing a docstring, only mentioning this since we've been consistent with adding docstrings to the rest of the actions and assertions.